scipy/optimize/_linprog.py

```diff
@@ -135,742 +141,1009 @@ def linprog_terse_callback(xk, **kwargs):
+    i_free = np.nonzero(l_free)[0]
+    n_free = len(i_free)
+    A1 = hstack((A1, zeros((A1.shape[0], n_free))))
+    c = np.concatenate((c, np.zeros(n_free)))
+    A1[:, range(n_ub, A1.shape[1])] = -A1[:, i_free]
+    c[np.arange(n_ub, A1.shape[1])] = -c[i_free]
+
+    # add slack variables
+    A2 = vstack([eye(A_ub.shape[0]), zeros((A_eq.shape[0], A_ub.shape[0]))])
+    A = hstack([A1, A2])
+
+    # lower bound: substitute xi = xi' + lb
+    # now there is a constant term in objective
+    i_shift = np.nonzero(lb_some)[0]
+    lb_shift = lbs[lb_some].astype(float)
+    c0 += np.sum(lb_shift * c[i_shift])
+    if sparse:
+        b = b.reshape(-1, 1)
+        A = A.tocsc()
+        b -= (A[:, i_shift] * sps.diags(lb_shift)).sum(axis=1)
+        b = b.ravel()
+    else:
+        b -= (A[:, i_shift] * lb_shift).sum(axis=1)
+
+    return A, b, c, c0
+
+
+def _postprocess(
+        x,
+        c,
+        A_ub=None,
+        b_ub=None,
+        A_eq=None,
+        b_eq=None,
+        bounds=None,
+        complete=False,
+        undo=[],
+        status=0,
+        message="",
+        tol=1e-8):
+    """
+    Given solution x to presolved, standard form linear program x, add
+    fixed variables back into the problem and undo the variable substitutions
+    to get solution to original linear program. Also, calculate the objective
+    function value, slack in original upper bound constraints, and residuals
+    in original equality constraints.
 
-    where:  -inf <= x[0] <= inf
+    Parameters
+    ----------
+    x : 1-D array
+        Solution vector to the standard-form problem.
+    c : 1-D array
+        Original coefficients of the linear objective function to be minimized.
+    A_ub : 2-D array
+        Original upper bound constraint matrix.
+    b_ub : 1-D array
+        Original upper bound constraint vector.
+    A_eq : 2-D array
+        Original equality constraint matrix.
+    b_eq : 1-D array
+        Original equality constraint vector.
+    bounds : sequence of tuples
+        Bounds, as modified in presolve
+    complete : bool
+        Whether the solution is was determined in presolve (``True`` if so)
+    undo: list of tuples
+        (`index`, `value`) pairs that record the original index and fixed value
+        for each variable removed from the problem
+    status : int
+        An integer representing the exit status of the optimization::
 
-    This problem deviates from the standard linear programming problem.  In
-    standard form, linear programming problems assume the variables x are
-    non-negative.  Since the variables don't have standard bounds where
-    0 <= x <= inf, the bounds of the variables must be explicitly set.
+             0 : Optimization terminated successfully
+             1 : Iteration limit reached
+             2 : Problem appears to be infeasible
+             3 : Problem appears to be unbounded
+             4 : Serious numerical difficulties encountered
```

Comment:
What is not "serious" in terms of difficulties?

Reply:
Difficulties that have been resolved using a more robust, albeit less efficient, solver.